tests/pr_test.go

```diff
@@ -18,6 +18,9 @@ func setupOptions(t *testing.T, prefix string, dir string) *testhelper.TestOptio
 		TerraformDir:  dir,
 		Prefix:        prefix,
 		ResourceGroup: resourceGroup,
+		TerraformVars: map[string]interface{}{
+			"prefix": prefix,
+		},
```

Comment:
As far as I know, you don't have to explicitly pass a prefix value here. The `TestOptionsDefault` function automatically will add a prefix

Reply:
Initially, I did not pass the prefix explicitly but it was failing in the pipeline throwing the following error.

>  Error: [ERROR] Error creating access group: An access group with the name basic-test-access-group already exists. Enter a different name.

On looking through the plan I observed that it is not taking up the prefix value from the `TestOptionsDefault` function that is why I added prefix explicitly.

Reply:
@toddgiguere Is this test misconfigured here?

Reply:
If you use `TestOptionsDefault`, then you will need to set all of the required variables, including "region", "prefix", "resource_group" and "resource_tags".

If you want the testwrapper to set the above variables for you based on your other options, then you should be using `TestOptionsDefaultWithVars`.

Glancing at this code, I'm thinking you should be using `TestOptionsDefaultWithVars` and remove the TerraformVars.

Reply:
@toddgiguere We can't use `TestOptionsDefaultWithVars` because the example does not take in a resource group variable. So seems we need to explicitly set all required vars in `TerraformVars` so. thanks for confirming

Reply:
Actually I'm not sure if we should be going with `"prefix": prefix,` - that means it will always use the same prefix value, instead of using the random one that the wrapper creates. Maybe we need to try this:

```
options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
	Testing:       t,
	TerraformDir:  dir,
	Prefix:        prefix,
	ResourceGroup: resourceGroup,
})

options.TerraformVars = map[string]interface{}{
	"prefix":           options.Prefix,
}
```